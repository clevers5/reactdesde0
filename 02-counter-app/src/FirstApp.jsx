import React from "react"


const Nombre = "Clevers Arnez"
const newMessage ={
  title: "Titulo del mensaje",
  id: 1,
  nombre: "Clevers Arnez",
}
const getResult = () => {
  return 3+3;
}

const Numeros = [1,2,3,4,5,6,7,8,9];
export const FirstApp = ({title, subtitle = "Soy un subtitulo"}) => {
  // console.log(props);
  
  return (
    
    //Fragment <> </> is the same as <React.Fragment> </React.Fragment>
    <React.Fragment>
        <h1>{title}</h1>
        <h2>{subtitle}</h2>
        
        {/* <h2>{newMessage.nombre}</h2>
        <h1>{getResult()}</h1> */}
        {/* <code>{JSON.stringify(newMessage)}</code> */}
        {/* <h1>Hola mundo</h1>
        <h3>{Numeros}</h3>
        <p>My first app</p> */}
    </React.Fragment>
    
  )
}
