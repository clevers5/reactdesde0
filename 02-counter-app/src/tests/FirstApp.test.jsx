import {render} from '@testing-library/react';
import {FirstApp} from '../FirstApp';

describe('Pruebas en FirstApp', () => { 
    test('Debe de hacer match con el snapshot', () => {
        const title = "Hola mundo";
        const { container } = render(<FirstApp title={title}/>);
         console.log(container);


         //Para hacer match con el snapshot se usa el siguiente comando:
        expect(container).toMatchSnapshot();
    })

    test('Debe de mostrar el titulo en un h1', () => {
        const title = "Hola mundo";
        const { container, getByText } = render(<FirstApp title={title}/>);
        
        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);
        //toContain sirve para buscar un texto dentro de un string sin importar los espacios
    })
 
})