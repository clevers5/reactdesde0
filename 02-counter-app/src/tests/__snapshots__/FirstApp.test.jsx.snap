// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola mundo
  </h1>
  <h2>
    Soy un subtitulo
  </h2>
  <h3>
    Clevers Arnez
  </h3>
</div>
`;
